### ISSUE TITLE
Graphql not working on fresh installation 2.4.7-p2

### ISSUE THREAD

**Issue by felek000** [magento/magento2#39213]
### Preconditions and environment

- Magento version
2.4.7-p2

### Steps to reproduce

I used warden to install localy m2. 
All pages loaded fine.
Admin works
Swagger page worx.

When i enter 
https://app.demo.test/graphql

I have error:
```
{
"errors": [
{
"message": "Syntax Error: Unexpected <EOF>",
"locations": [
{
"line": 1,
"column": 1
}
],
"extensions": {
"file": "[/var/www/html/vendor/webonyx/graphql-php/src/Language/Parser.php](https://app.demo.test/var/www/html/vendor/webonyx/graphql-php/src/Language/Parser.php)",
"line": 433
}
}
]
}
```

in magento 2 exeption

```
Next Exception: Report ID: graph-ql-66f3e9cea655b; Message: Syntax Error: Unexpected <EOF> in /var/www/html/vendor/magento/framework/GraphQl/Exception/ExceptionFormatter.php:64
Stack trace:
#0 /var/www/html/vendor/magento/module-graph-ql/Controller/GraphQl.php(210): Magento\Framework\GraphQl\Exception\ExceptionFormatter->create(Object(GraphQL\Error\SyntaxError))
#1 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(58): Magento\GraphQl\Controller\GraphQl->dispatch(Object(Magento\Framework\App\Request\Http))
#2 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(138): Magento\GraphQl\Controller\GraphQl\Interceptor->___callParent('dispatch', Array)
#3 /var/www/html/vendor/magento/module-page-cache/Model/App/FrontController/BuiltinPlugin.php(72): Magento\GraphQl\Controller\GraphQl\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#4 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(135): Magento\PageCache\Model\App\FrontController\BuiltinPlugin->aroundDispatch(Object(Magento\GraphQl\Controller\GraphQl\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#5 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(153): Magento\GraphQl\Controller\GraphQl\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#6 /var/www/html/generated/code/Magento/GraphQl/Controller/GraphQl/Interceptor.php(23): Magento\GraphQl\Controller\GraphQl\Interceptor->___callPlugins('dispatch', Array, Array)
#7 /var/www/html/vendor/magento/framework/App/Http.php(116): Magento\GraphQl\Controller\GraphQl\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#8 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\App\Http->launch()
#9 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\App\Http\Interceptor->___callParent('launch', Array)
#10 /var/www/html/vendor/magento/module-application-performance-monitor/Plugin/ApplicationPerformanceMonitor.php(38): Magento\Framework\App\Http\Interceptor->Magento\Framework\Interception\{closure}()
#11 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(135): Magento\ApplicationPerformanceMonitor\Plugin\ApplicationPerformanceMonitor->aroundLaunch(Object(Magento\Framework\App\Http\Interceptor), Object(Closure))
#12 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\App\Http\Interceptor->Magento\Framework\Interception\{closure}()
#13 /var/www/html/generated/code/Magento/Framework/App/Http/Interceptor.php(23): Magento\Framework\App\Http\Interceptor->___callPlugins('launch', Array, NULL)
#14 /var/www/html/vendor/magento/framework/App/Bootstrap.php(264): Magento\Framework\App\Http\Interceptor->launch()
#15 /var/www/html/pub/index.php(30): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http\Interceptor))
#16 {main} {"exception":"[object] (Exception(code: 0): Report ID: graph-ql-66f3e9cea655b; Message: Syntax Error: Unexpected <EOF> at /var/www/html/vendor/magento/framework/GraphQl/Exception/ExceptionFormatter.php:64)
[previous exception] [object] (GraphQL\\Error\\SyntaxError(code: 0): Syntax Error: Unexpected <EOF> at /var/www/html/vendor/webonyx/graphql-php/src/Language/Parser.php:433)"} []
```

### Expected result

Page loads proper

### Actual result

```
Page loads and i can test query{
"errors": [
{
"message": "Syntax Error: Unexpected <EOF>",
"locations": [
{
"line": 1,
"column": 1
}
],
"extensions": {
"file": "[/var/www/html/vendor/webonyx/graphql-php/src/Language/Parser.php](https://app.demo.test/var/www/html/vendor/webonyx/graphql-php/src/Language/Parser.php)",
"line": 433
}
}
]
}
```

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Bravo** [comment]
Hi @felek000,

Thanks for your reporting and collaboration.

Could you please let us know for which of the mutation you are facing an issue.Could you please follow the document https://developer.adobe.com/commerce/webapi/graphql/usage/ for the graphql usage and let us know if you are still facing any issue.

Thanks.

**Comment by engcom-Bravo** [comment]
Hi @felek000,

Thanks for your update.

We have to run with the query or mutation with the endpoint otherwise it will throw an syntax error and it is expected behaviour.Since it was working fine with the query/mutation we are closing this issue.

Thanks.

**Comment by chandreshcp** [comment]
Hi @engcom-Bravo 

I'm also facing the same issue on the 2.4.6-p11.

When we run the graphql with the correct query It's working fine. But when we just https://www.example.com/graphql without any query or incorrect query giving error as below.

GraphQL\Error\SyntaxError: Syntax Error: Unexpected <EOF> in /var/www/html/vendor/webonyx/graphql-php/src/Language/Parser.php:437

In this case it's giving 500 errors. 

Here exception should be handled correctly and should not give 500 error.

please check the issue.

**Comment by nhp** [comment]
Sadly there is only an available quality patch for `=>2.4.7`.

**Comment by chandreshcp** [comment]
Hi @nhp 

Yes I have checked that. For this issue patch is available ACSD-63325 but it's not compatible for 2.4.6-p11.

@engcom-Bravo Any other patch compatible for 2.4.6-p11?
